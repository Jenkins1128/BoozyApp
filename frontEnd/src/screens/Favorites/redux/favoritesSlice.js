import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
