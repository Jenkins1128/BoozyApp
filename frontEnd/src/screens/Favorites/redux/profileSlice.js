import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
