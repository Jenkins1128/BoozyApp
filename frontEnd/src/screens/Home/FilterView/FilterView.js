import React from 'react';
import { View, Animated, Text, TextInput, TouchableOpacity, TouchableWithoutFeedback, StyleSheet, Keyboard } from 'react-native';

const FilterView = ({ bounceValue, dispatch, state, reset, updateLocation, updateCuisine, setPrice1, setPrice2, setPrice3, setPrice4, showFilterOverlay, getDataFromFilter }) => {
	return (
		<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
			<Animated.View style={[styles.subView, { transform: [{ translateY: bounceValue }] }]}>
				<View style={styles.header}>
					<View style={styles.title}>
						<Text style={{ fontSize: 20, fontWeight: 'bold', color: '#EB8873' }}>Filters</Text>
					</View>
					<View style={styles.headerSettings}>
						<TouchableOpacity onPress={() => dispatch(reset())} style={styles.resetButton}>
							<Text style={styles.resetText}>Reset</Text>
						</TouchableOpacity>
						<TouchableOpacity onPress={showFilterOverlay} style={styles.cancelButton}>
							<Text style={styles.cancelText}>Cancel</Text>
						</TouchableOpacity>
					</View>
				</View>
				<View style={styles.filterInput}>
					<TextInput
						style={styles.locationStyle}
						textAlign='center'
						multiline={false}
						returnKeyType='next'
						onChangeText={(location) => dispatch(updateLocation({ location: location }))}
						value={state.location}
						placeholder='Enter City or Zip code'
					/>
					<TextInput style={styles.cuisineStyle} textAlign='center' multiline={false} returnKeyType='next' onChangeText={(cuisine) => dispatch(updateCuisine({ cuisine: cuisine }))} value={state.cuisine} placeholder='Enter cuisine' />
					<View style={styles.priceTypeContainer}>
						<TouchableOpacity onPress={() => dispatch(setPrice1())} style={styles.$Button}>
							<Text style={{ color: state.$color, fontWeight: 'bold', fontSize: 16 }}>$</Text>
						</TouchableOpacity>
						<TouchableOpacity onPress={() => dispatch(setPrice2())} style={styles.$$Button}>
							<Text style={{ color: state.$$color, fontWeight: 'bold', fontSize: 16 }}>$$</Text>
						</TouchableOpacity>
						<TouchableOpacity onPress={() => dispatch(setPrice3())} style={styles.$$$Button}>
							<Text style={{ color: state.$$$color, fontWeight: 'bold', fontSize: 16 }}>$$$</Text>
						</TouchableOpacity>
						<TouchableOpacity onPress={() => dispatch(setPrice4())} style={styles.$$$$Button}>
							<Text style={{ color: state.$$$$color, fontWeight: 'bold', fontSize: 16 }}>$$$$</Text>
						</TouchableOpacity>
					</View>
					<TouchableOpacity onPress={getDataFromFilter} style={styles.applyButton}>
						<Text style={styles.applyText}>Apply</Text>
					</TouchableOpacity>
				</View>
			</Animated.View>
		</TouchableWithoutFeedback>
	);
};

export default FilterView;

const styles = StyleSheet.create({
	subView: {
		position: 'absolute',
		backgroundColor: '#FFFFFF',
		padding: 20,
		borderRadius: 30
	},
	header: {
		flex: 1
	},
	title: {
		flex: 1,
		alignSelf: 'center'
	},
	headerSettings: {
		flexDirection: 'row',
		justifyContent: 'space-between'
	},
	filterInput: {
		flex: 3,
		justifyContent: 'center',
		alignItems: 'center'
	},

	locationStyle: {
		flex: 1,
		padding: 20,
		fontSize: 18
	},
	cuisineStyle: {
		flex: 1,
		padding: 20,
		fontSize: 18
	},
	priceTypeContainer: {
		flexDirection: 'row',
		justifyContent: 'space-around',
		height: 100,
		width: '80%'
	},
	textInput: {
		alignSelf: 'stretch',
		color: '#fff',
		padding: 20,
		backgroundColor: '#252525',
		borderTopWidth: 2,
		borderTopColor: '#ededed'
	},
	addButton: {
		backgroundColor: '#E91E63',
		width: 90,
		height: 90,
		borderRadius: 50,
		alignItems: 'center',
		justifyContent: 'center',
		elevation: 8
	},
	addButtonText: {
		color: '#fff',
		fontSize: 24
	},
	cancelButton: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#FFFFFF',
		padding: 10,
		borderRadius: 20
	},
	resetButton: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#FFFFFF',
		padding: 10,
		borderRadius: 20
	},
	applyButton: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		padding: 10,
		borderRadius: 20
	},
	$Button: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		borderRadius: 30,
		width: 45,
		height: 45
	},
	$$Button: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		borderRadius: 20,
		borderRadius: 30,
		width: 45,
		height: 45
	},
	$$$Button: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		borderRadius: 20,
		borderRadius: 30,
		width: 45,
		height: 45
	},
	$$$$Button: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		borderRadius: 20,
		borderRadius: 30,
		width: 45,
		height: 45
	},
	resetText: {
		color: '#EB8873',
		fontWeight: 'bold'
	},
	cancelText: {
		color: '#EB8873',
		fontWeight: 'bold'
	},
	applyText: {
		color: 'white',
		fontWeight: 'bold'
	}
});
