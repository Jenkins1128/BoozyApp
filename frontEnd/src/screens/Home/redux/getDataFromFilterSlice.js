import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
