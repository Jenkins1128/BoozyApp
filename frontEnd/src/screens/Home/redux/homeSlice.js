import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${latitude}/${longitude}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState.location = payload.location;
		},
		updateCuisine: (state, { payload }) => {
			state.currentState.cuisine = payload.cuisine;
		},
		updatePriceType: (state, { payload }) => {
			state.currentState.priceType = payload.priceType;
		},
		update$color: (state, { payload }) => {
			state.currentState.$color = payload.$color;
		},
		update$$color: (state, { payload }) => {
			state.currentState.$$color = payload.$$color;
		},
		update$$$color: (state, { payload }) => {
			state.currentState.$$$color = payload.$$$color;
		},
		update$$$$color: (state, { payload }) => {
			state.currentState.$$$$color = payload.$$$$color;
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState.allCategories = payload.allCategories;
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState.alreadyFavorited = payload.alreadyFavorited;
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState.alreadyRated = payload.alreadyRated;
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState.restaurantsArray = payload.restaurantsArray;
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState.initialPosition = payload.initialPosition;
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState.markerPosition = payload.markerPosition;
		},
		reset: (state) => {
			Object.assign(state.currentState, {
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice1: (state) => {
			Object.assign(state.currentState, {
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice2: (state) => {
			Object.assign(state.currentState, {
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice3: (state) => {
			Object.assign(state.currentState, {
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			});
		},
		setPrice4: (state) => {
			Object.assign(state.currentState, {
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			});
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					locationDataRequestStatus: 'fulfilled',
					restaurantsArray: payload
				});
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState.locationDataRequestStatus = 'rejected';
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
