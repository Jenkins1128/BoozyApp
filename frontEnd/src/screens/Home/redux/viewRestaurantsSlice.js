import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const viewRestaurantsAsync = createAsyncThunk('viewRestaurantsAsync/status', async (data, { rejectWithValue }) => {
	const { index, state } = data;
	state.index = index;
	const restaurantId = state.restaurantsArray[index]['id'];
	const name = state.restaurantsArray[index]['name'];
	try {
		const alreadyFavoritedResponse = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/favorites', {
			restaurantId: restaurantId,
			name: name
		});
		state.alreadyFavorited = alreadyFavoritedResponse.data[0]['contains'];
		const alreadyRatedResponse = await axios.get(`https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/${state.restaurantsArray[index]['id']}/review`);
		state.alreadyRated = alreadyRatedResponse.data[0]['contains'];
		state.allCategories = '';
		for (let i = 0; i < state.restaurantsArray[index]['categories'].length; i++) {
			if (i == 0) {
				state.allCategories += state.restaurantsArray[index]['categories'][i]['title'];
			} else {
				state.allCategories += ', ' + state.restaurantsArray[index]['categories'][i]['title'];
			}
		}
		const newState = { ...state, restaurantsArray: [state.restaurantsArray[index]] };
		return newState;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const viewRestaurantsSlice = createSlice({
	name: 'viewRestaurantsAsync',
	initialState,
	reducers: {
		resetRestaurantRequestStatus: (state) => {
			state.currentState.restaurantRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(viewRestaurantsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantInfo: payload, restaurantRequestStatus: 'fulfilled' };
			})
			.addCase(viewRestaurantsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantRequestStatus: 'rejected' };
			});
	}
});

export const { resetRestaurantRequestStatus } = viewRestaurantsSlice.actions;
export const selectRestaurantState = (state) => state.restaurant.currentState;

export default viewRestaurantsSlice.reducer;
