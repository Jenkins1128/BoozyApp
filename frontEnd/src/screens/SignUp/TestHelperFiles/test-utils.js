import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
