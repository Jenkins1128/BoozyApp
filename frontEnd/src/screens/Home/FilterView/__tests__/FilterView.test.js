import React from 'react';
import { cleanup, render, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';
import FilterView from '../FilterView';

describe('<FilterView />', () => {
	let state;
	let dispatch, reset, updateLocation, updateCuisine, setPrice1, setPrice2, setPrice3, setPrice4, showFilterOverlay, getDataFromFilter;
	let rendered;

	beforeEach(() => {
		state = {
			location: 'France',
			cuisine: 'pizza',
			$color: 'white',
			$$color: 'white',
			$$$color: 'white',
			$$$$color: 'white'
		};
		(dispatch = jest.fn()),
			(reset = jest.fn()),
			(updateLocation = jest.fn()),
			(updateCuisine = jest.fn()),
			(setPrice1 = jest.fn()),
			(setPrice2 = jest.fn()),
			(setPrice3 = jest.fn()),
			(setPrice4 = jest.fn()),
			(showFilterOverlay = jest.fn()),
			(getDataFromFilter = jest.fn());
		rendered = render(
			<FilterView
				dispatch={dispatch}
				state={state}
				reset={reset}
				updateLocation={updateLocation}
				updateCuisine={updateCuisine}
				setPrice1={setPrice1}
				setPrice2={setPrice2}
				setPrice3={setPrice3}
				setPrice4={setPrice4}
				showFilterOverlay={showFilterOverlay}
				getDataFromFilter={getDataFromFilter}
			/>
		);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should render subView styles correctly', () => {
		const subViewComponent = rendered.getByTestId('subView');
		const given = subViewComponent.props.style;
		const result = {
			position: 'absolute',
			backgroundColor: '#FFFFFF',
			padding: 20,
			borderRadius: 30,
			transform: [{ translateY: undefined }]
		};
		expect(result).toMatchObject(given);
	});

	it('should render header styles correctly', () => {
		const headerComponent = rendered.getByTestId('header');
		const given = headerComponent.props.style;
		const result = {
			flex: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render title styles correctly', () => {
		const titleComponent = rendered.getByTestId('title');
		const given = titleComponent.props.style;
		const result = {
			flex: 1,
			alignSelf: 'center'
		};
		expect(result).toMatchObject(given);
	});

	it('should render headerSettings styles correctly', () => {
		const headerSettingsComponent = rendered.getByTestId('headerSettings');
		const given = headerSettingsComponent.props.style;
		const result = {
			flexDirection: 'row',
			justifyContent: 'space-between'
		};
		expect(result).toMatchObject(given);
	});

	it('should render filterInput styles correctly', () => {
		const filterInputComponent = rendered.getByTestId('filterInput');
		const given = filterInputComponent.props.style;
		const result = {
			flex: 3,
			justifyContent: 'center',
			alignItems: 'center'
		};
		expect(result).toMatchObject(given);
	});

	it('should render locationInput styles correctly', () => {
		const locationInputComponent = rendered.getByTestId('locationInput');
		const given = locationInputComponent.props.style;
		const result = {
			flex: 1,
			padding: 20,
			fontSize: 18
		};
		expect(result).toMatchObject(given);
	});

	it('should render cuisineInput styles correctly', () => {
		const cuisineInputComponent = rendered.getByTestId('cuisineInput');
		const given = cuisineInputComponent.props.style;
		const result = {
			flex: 1,
			padding: 20,
			fontSize: 18
		};
		expect(result).toMatchObject(given);
	});

	it('should render priceTypeContainer styles correctly', () => {
		const priceTypeContainerComponent = rendered.getByTestId('priceTypeContainer');
		const given = priceTypeContainerComponent.props.style;
		const result = {
			flexDirection: 'row',
			justifyContent: 'space-around',
			height: 100,
			width: '80%'
		};
		expect(result).toMatchObject(given);
	});

	it('should render cancelButton styles correctly', () => {
		const cancelButtonComponent = rendered.getByTestId('cancelButton');
		const given = cancelButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#FFFFFF',
			padding: 10,
			borderRadius: 20,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render resetButton styles correctly', () => {
		const resetButtonComponent = rendered.getByTestId('resetButton');
		const given = resetButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#FFFFFF',
			padding: 10,
			borderRadius: 20,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render applyButton styles correctly', () => {
		const applyButtonComponent = rendered.getByTestId('applyButton');
		const given = applyButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			padding: 10,
			borderRadius: 20,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render $Button styles correctly', () => {
		const $ButtonComponent = rendered.getByTestId('$Button');
		const given = $ButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			borderRadius: 30,
			width: 45,
			height: 45,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render $$Button styles correctly', () => {
		const $$ButtonComponent = rendered.getByTestId('$$Button');
		const given = $$ButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			borderRadius: 20,
			borderRadius: 30,
			width: 45,
			height: 45,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render $$$Button styles correctly', () => {
		const $$$ButtonComponent = rendered.getByTestId('$$$Button');
		const given = $$$ButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			borderRadius: 20,
			borderRadius: 30,
			width: 45,
			height: 45,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render $$$$Button styles correctly', () => {
		const $$$$ButtonComponent = rendered.getByTestId('$$$$Button');
		const given = $$$$ButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			borderRadius: 20,
			borderRadius: 30,
			width: 45,
			height: 45,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render resetText styles correctly', () => {
		const resetTextComponent = rendered.getByTestId('resetText');
		const given = resetTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render cancelText styles correctly', () => {
		const cancelTextComponent = rendered.getByTestId('cancelText');
		const given = cancelTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render applyText styles correctly', () => {
		const applyTextComponent = rendered.getByTestId('applyText');
		const given = applyTextComponent.props.style;
		const result = {
			color: 'white',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire filterView location input onChangeText events', () => {
		const locationInputComponent = rendered.getByTestId('locationInput');
		fireEvent(locationInputComponent, 'changeText', 'new text');
		expect(updateLocation).toHaveBeenCalledWith({ location: 'new text' });
	});

	it('should fire filterView cuisine input onChangeText events', () => {
		const updateCuisineComponent = rendered.getByTestId('cuisineInput');
		fireEvent(updateCuisineComponent, 'changeText', 'new text');
		expect(updateCuisine).toHaveBeenCalledWith({ cuisine: 'new text' });
	});
});
