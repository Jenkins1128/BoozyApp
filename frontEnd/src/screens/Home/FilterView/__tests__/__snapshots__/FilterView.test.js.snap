// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterView /> should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 30,
      "padding": 20,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": undefined,
        },
      ],
    }
  }
  testID="subView"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="header"
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
        }
      }
      testID="title"
    >
      <Text
        style={
          Object {
            "color": "#EB8873",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Filters
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      testID="headerSettings"
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="resetButton"
      >
        <Text
          style={
            Object {
              "color": "#EB8873",
              "fontWeight": "bold",
            }
          }
          testID="resetText"
        >
          Reset
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="cancelButton"
      >
        <Text
          style={
            Object {
              "color": "#EB8873",
              "fontWeight": "bold",
            }
          }
          testID="cancelText"
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
    testID="filterInput"
  >
    <TextInput
      allowFontScaling={true}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Enter City or Zip code"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "flex": 1,
          "fontSize": 18,
          "padding": 20,
        }
      }
      testID="locationInput"
      textAlign="center"
      underlineColorAndroid="transparent"
      value="France"
    />
    <TextInput
      allowFontScaling={true}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Enter cuisine"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "flex": 1,
          "fontSize": 18,
          "padding": 20,
        }
      }
      testID="cuisineInput"
      textAlign="center"
      underlineColorAndroid="transparent"
      value="pizza"
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "space-around",
          "width": "80%",
        }
      }
      testID="priceTypeContainer"
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EB8873",
            "borderRadius": 30,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": 45,
          }
        }
        testID="$Button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          $
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EB8873",
            "borderRadius": 30,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": 45,
          }
        }
        testID="$$Button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          $$
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EB8873",
            "borderRadius": 30,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": 45,
          }
        }
        testID="$$$Button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          $$$
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EB8873",
            "borderRadius": 30,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": 45,
          }
        }
        testID="$$$$Button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          $$$$
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EB8873",
          "borderRadius": 20,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="applyButton"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
        testID="applyText"
      >
        Apply
      </Text>
    </View>
  </View>
</View>
`;
